teps/0139-trusted-artifacts.md

```diff
@@ -0,0 +1,478 @@
+                else
+                  tar zcf download.tgz $(params.aFolderArtifact.path)
+                  echo "$(params.aFolderArtifact.hash) download.tgz" | md5sum -c
+                fi
+            - name: consume-content
+              image: bash:latest
+              script: |
+                #!/usr/bin/env bash
+                # Do something with the verified content
+                # Here I need to use a workspace variable to trigger propagation of the workspace
+                find $(workspaces.artifactStorage.path) -type f
+        params:
+          - name: aFileArtifact
+            value: $(tasks.producer.results.aFileArtifact)
+          - name: aFolderArtifact
+            value: $(tasks.producer.results.aFolderArtifact)
+  workspaces:
+  - name: artifactStorage
+    volumeClaimTemplate:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+```
+</details>
+
+</br>
+
+<details>
+  <summary>Example execution log:</summary>
+
+```log
+[producer : produce-file] 1690234279
+
+[producer : upload-hash-file] {
+[producer : upload-hash-file]   "path": "/workspace/artifactStorage/afile.txt",
+[producer : upload-hash-file]   "hash": "77c5df93c80c4847891407f22c955527",
+[producer : upload-hash-file]   "type": "file"
+[producer : upload-hash-file] }
+
+[producer : produce-folder] 1690234281
+[producer : produce-folder] 1690234281
+[producer : produce-folder] 1690234281
+
+[producer : upload-hash-folder] tar: removing leading '/' from member names
+[producer : upload-hash-folder] {
+[producer : upload-hash-folder]   "path": "/workspace/artifactStorage/afolder",
+[producer : upload-hash-folder]   "hash": "ce344e90cd05a43e44db451dc9d91354",
+[producer : upload-hash-folder]   "type": "folder"
+[producer : upload-hash-folder] }
+
+[consumer : download-verify-file] /workspace/artifactStorage/afile.txt: OK
+
+[consumer : download-verify-folder] tar: removing leading '/' from member names
+[consumer : download-verify-folder] download.tgz: OK
+
+[consumer : consume-content] /workspace/artifactStorage/afile.txt
+[consumer : consume-content] /workspace/artifactStorage/afolder/a.txt
+[consumer : consume-content] /workspace/artifactStorage/afolder/b.txt
+[consumer : consume-content] /workspace/artifactStorage/afolder/c.txt
+```    
+</details>
+
+</br>
+
+
+The example pipeline shows a few things:
+- The solution of sharing content is generic, there's nothing in it specific to pipeline in question
+- Already in the pipeline the same code is used more than once. Replicating this solution in a pipeline with multiple producer or consumers would lead to a lot of duplication
+- The metadata (path, hash and type) required to trust an artifact is stored as a result in the status of the `TaskRun`. To complete the chain of trust we need to be able to trust the `status` of the `TaskRun`, feature that would be provided by the integration with Spire proposed in TEP-0089
```

Comment:
Good point, I guess it depends on how artifacts stored in alternative storage are implemented... I considered path to be relative to the storage, so for now the path within the PVC, in future it could be the path in an object storage or else. The URI to the storage should be configured elsewhere,